segmentstore/server/src/test/java/io/pravega/segmentstore/server/logs/DurableLogTests.java

```diff
@@ -1071,10 +1072,12 @@ public void testTruncateWithoutRecovery() {
             // recovery, it wipes away all existing metadata).
             Set<Long> streamSegmentIds = createStreamSegmentsWithOperations(streamSegmentCount, durableLog);
             List<Operation> queuedOperations = generateOperations(streamSegmentIds, new HashMap<>(), appendsPerStreamSegment, METADATA_CHECKPOINT_EVERY, false, false);
-            queuedOperations.add(new MetadataCheckpointOperation()); // Add one of these at the end to ensure we can truncate everything.
+            val lastOp = new MetadataCheckpointOperation();
+            queuedOperations.add(lastOp); // Add one of these at the end to ensure we can truncate everything.
 
             List<OperationWithCompletion> completionFutures = processOperations(queuedOperations, durableLog);
             OperationWithCompletion.allOf(completionFutures).join();
+            awaitLastOperationAdded(durableLog, metadata);
```

Comment:
First one: we need to read all operations in the log, and we want to make sure all are accounted for.
Second one: we await the `StorageMetadataCheckpointOperation` that is added during a truncation because we then need to read it from the log.
Third one: we add a `MetadataCheckpointOperation` and then need to truncate on it. As such, we need that to be present in the log before truncating.